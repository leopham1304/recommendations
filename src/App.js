import React, { memo } from "react";
import Routes from "./routes";

const AppComponent = () => {
  return (
    <div className="App">
      <Routes />
    </div>
  );
};

const App = memo(AppComponent);
App.displayName = "App";
export default App;
