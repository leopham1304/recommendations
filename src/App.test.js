import React from "react";
import App from "./App.js";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

describe("App", () => {
  const wrapper = mount(<App />);

  it("should render App without crashing", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
