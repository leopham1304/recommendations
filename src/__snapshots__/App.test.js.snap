// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App without crashing 1`] = `
<App>
  <div
    className="App"
  >
    <Memo(Routes)>
      <BrowserRouter>
        <Router
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <Switch>
            <Route
              computedMatch={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              exact={true}
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              path="/"
            >
              <Memo(Landing)>
                <div
                  className="landing-container"
                >
                  <div
                    className="introduction-wrapper"
                  >
                    <h1>
                      Hello, welcome to Vietnam!
                    </h1>
                    <div
                      className="introduction-text"
                    >
                      <p>
                        Vietnam is a country divided into 58 provinces
                      </p>
                      <p>
                        There are many heavenly breathtaking landscapes to explore
                      </p>
                    </div>
                    <button
                      className="landing-button"
                      onClick={[Function]}
                    >
                      Fly me to the 
                      <strong>
                         Moon 
                      </strong>
                    </button>
                  </div>
                </div>
              </Memo(Landing)>
            </Route>
          </Switch>
        </Router>
      </BrowserRouter>
    </Memo(Routes)>
  </div>
</App>
`;
